### PR TITLE
DS-3184: Add Aqua GHA

### PR DESCRIPTION
**Pull request by radobar**
**What this PR does / why we need it**:
Expands the workflow with aqua GHA and refactors it a bit.

**Which issue(s) this PR fixes** (optional)
Closes [#DS-3184](https://portworx.atlassian.net/browse/DS-3184)

### CODE REVIEW

File: .github/workflows/image-scan.yml [hunk 1]
```diff
@@ -102,109 +135,146 @@ jobs:
...
+            statusIcon=":green_circle:"
+            for jsonFile in $lwResultFiles; do
+              cveC=$(jq -r '.stats.Critical.total' $jsonFile)
+              cveH=$(jq -r '.stats.High.total' $jsonFile)
+              if [ $cveC -gt 0 ] || [ $cveH -gt 0 ]; then
+                lwStatusIcon=":red_circle:"
+                break
+              fi
+            done       
+          }
```
Comment (jsimonps) [inline .github/workflows/image-scan.yml:223]: It looks like `setStatusIcon()` is still hardcoded to process Lacework results only.

Comment (radobar) [reply .github/workflows/image-scan.yml:223]: Thanks, nice catch, missed it.

File: .github/actions/action-aqua-scan/action.yml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        jq '{
+              image: "${{ inputs.image }}",
+              stats: {
+                Critical: { total: (if .vulnerability_summary.critical == null then 0 else .vulnerability_summary.critical end) },
+                High: { total: (if .vulnerability_summary.high == null then 0 else .vulnerability_summary.high end) },
+                Medium: { total: (if .vulnerability_summary.medium == null then 0 else .vulnerability_summary.medium end) },
+                Low: { total: (if .vulnerability_summary.low == null then 0 else .vulnerability_summary.low end) },
+                Info: { total: (if .vulnerability_summary.negligible == null then 0 else .vulnerability_summary.negligible end) },
+              },
+              includedSeverities: [ "Critical", "High" ],
```
Comment (jsimonps) [inline .github/actions/action-aqua-scan/action.yml:112]: I see you copied my unfinished filter :smile:  Maybe we could add a TODO comment into both scanners that the `severity-filter` is hard-coded yet.

File: .github/workflows/image-scan.yml (hunk 1 above)
Comment (jsimonps) [reply .github/workflows/image-scan.yml:223]: I can still see problem in using the `lwResultFiles` in `setStatusIcon()`, maybe just rename to `resultFiles`

**Review by jsimonps — APPROVED** [review]
LGTM, except the issue with the `setStatusIcon()`. Nice, thank you!

Don't forget to release a new version for it as due to new required parameters is not backward compatible.

File: .github/actions/action-aqua-scan/action.yml (hunk 2 above)
Comment (radobar) [reply .github/actions/action-aqua-scan/action.yml:112]: Yeah, I had a todo there, that I removed =D was thinking, that it's not really that useful/necessary for now, but we could keep it =D

File: .github/workflows/image-scan.yml [hunk 3]
```diff
@@ -102,109 +136,146 @@ jobs:
...
+      - name: Build PR comment
+        if: github.event_name == 'pull_request' && (github.event.action == 'opened' || github.event.action == 'synchronize')
+        id: build-pr-comment
+        shell: bash
+        run: |
+          createPRCommentTable() {
+            comment="${comment}\n| Image | Critical | High | Medium | Low | Info |"
+            comment="${comment}\n|---|---|---|---|---|---|"
+            
+            for resultJSONFile in $aquaResultFiles; do
```
Comment (jsimonps) [inline .github/workflows/image-scan.yml]: Rado, have you also tested these changes? :smile: 
```suggestion
            for resultJSONFile in $1; do
```

Comment (radobar) [reply .github/workflows/image-scan.yml]: Originally, yes, not the new ones, that's why I converted it to a draft =D testing atm in a private repo.

Comment (radobar) [reply .github/workflows/image-scan.yml]: Last minute changes were a bit careless from me, I admit :sweat:

Comment (jsimonps) [reply .github/workflows/image-scan.yml]: Ok, don't worry, we are almost there :smile:

**Comment by RadoBart** [comment]
redgreen
-4c95-918b-ff9a8e422e16.png)

**Comment by RadoBart** [comment]
redgreen
multiple

**Comment by radobar** [comment]
Rerequesting @jsimonps due to valuable points. Hopefully I fixed it :sweat_smile: . Added new tag due to backwards incompatible changes.

**Comment by radobar** [comment]
redgreen
multiple

**Review by jsimonps — APPROVED** [review]
LGTM, thank you! :+1:

**Pull request merged**